src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/UDPCommunicator.java

```diff
@@ -0,0 +1,493 @@
+	 *
+	 * @return List<String> This returns the current list of command error
+	 */
+	public List<String> getCommandErrorList() {
+		return this.commandErrorList;
+	}
+
+	/**
+	 * This method is used set the list of command error
+	 *
+	 * @param commandErrorList This is the list of command error to set
+	 */
+	protected void setCommandErrorList(List<String> commandErrorList) {
+		if (this.isInitialized()) {
+			throw new IllegalStateException("Cannot change properties after init() was called");
+		} else {
+			this.commandErrorList = commandErrorList;
+		}
+	}
+
+	/**
+	 * This method returns the list of command success
+	 *
+	 * @return List<String> This returns the current list of command success
+	 */
+	public List<String> getCommandSuccessList() {
+		return this.commandSuccessList;
+	}
+
+	/**
+	 * This method is used set the list of command success
+	 *
+	 * @param commandSuccessList This is the list of command success to set
+	 */
+	protected void setCommandSuccessList(List<String> commandSuccessList) {
+		if (this.isInitialized()) {
+			throw new IllegalStateException("Cannot change properties after init() was called");
+		} else {
+			this.commandSuccessList = commandSuccessList;
+		}
+	}
+
+	/**
+	 * This method is used to create a connection
+	 */
+	@Override
+	public void connect() {
+		if (!this.isInitialized()) {
+			throw new IllegalStateException("UDPCommunicator cannot be used before init() is called");
+		} else {
+			if (this.logger.isTraceEnabled()) {
+				this.logger.trace("Connecting to: " + this.host + " port: " + this.port);
+			}
+
+			Lock writeLock = this.lock.writeLock();
+			writeLock.lock();
+
+			try {
+				if (!this.isChannelConnected()) {
+					this.createChannel();
+					this.status.setLastTimestamp(System.currentTimeMillis());
+					this.status.setConnectionState(ConnectionState.Connected);
+					this.status.setLastError(null);
+				}
+			} catch (Exception ex) {
+				if (this.logger.isErrorEnabled()) {
+					this.logger.error("Error connecting to: " + this.host + " port: " + this.port, ex);
+				}
+
+				this.status.setLastError(ex);
+				this.status.setConnectionState(ConnectionState.Failed);
+				this.destroyChannel();
+				throw ex;
+			} finally {
+				writeLock.unlock();
+			}
+		}
+	}
+
+	/**
+	 * This method is used to disconnect from the device
+	 */
+	@Override
+	public void disconnect() {
+		if (this.logger.isTraceEnabled()) {
+			this.logger.trace("Disconnecting from: " + this.host + " port: " + this.port);
+		}
+
+		Lock writeLock = this.lock.writeLock();
+		writeLock.lock();
+
+		try {
+			this.destroyChannel();
+			this.status.setConnectionState(ConnectionState.Disconnected);
+		} finally {
+			writeLock.unlock();
+		}
+	}
+
+	/**
+	 * This method is used to get current connection status from the device
+	 */
+	@Override
+	public ConnectionStatus getConnectionStatus() {
+		Lock readLock = this.lock.readLock();
+		readLock.lock();
+
+		ConnectionStatus status;
```

Comment:
The `status` variable name is already defined in 
`private final ConnectionStatus status = new ConnectionStatus();`

I think we should use a different name to avoid the bug in the future.

Reply:
ok, I'll update it